File: iosApp/iosApp/Presentation/Main/CafeManagement/Schedule/ScheduleAction.swift
//
//  ScheduleAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/11.
//

import Foundation
import Shared

enum ScheduleAction {
    case clickBack
    case clickMore
    case clickCalendar
    case selectDay(id: String)
    case clickEditDay(id: String)
    case dismissEditSheet
    case changeEditStatus(CastScheduleStatus)
    case changeEditStartTime(String)
    case changeEditEndTime(String)
    case submitEditDay
    case clickSave
    case dismissInfoMessage
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/cafemanagement/schedule/ScheduleAction.kt
package com.hhp227.concafe.presentation.main.cafemanagement.schedule

import com.hhp227.concafe.domain.model.CastScheduleStatus

sealed interface ScheduleAction {
    data object ClickBack : ScheduleAction
    data object ClickMore : ScheduleAction
    data object ClickCalendar : ScheduleAction
    data class SelectPeriod(val period: SchedulePeriod) : ScheduleAction
    data class SelectDay(val dayId: String) : ScheduleAction
    data class ClickEditDay(val dayId: String) : ScheduleAction
    data object DismissEditSheet : ScheduleAction
    data class ChangeEditStatus(val status: CastScheduleStatus) : ScheduleAction
    data class ChangeEditStartTime(val value: String) : ScheduleAction
    data class ChangeEditEndTime(val value: String) : ScheduleAction
    data object SubmitEditDay : ScheduleAction
    data object ClickSave : ScheduleAction
    data object DismissInfoMessage : ScheduleAction
}
